lib/cc/engine/analyzers/violation.rb

```diff
@@ -1,52 +1,32 @@
-require "cc/engine/analyzers/sexp_lines"
-require "cc/engine/analyzers/violation_read_up"
-require "digest"
+require "cc/engine/analyzers/issue"
 
 module CC
   module Engine
     module Analyzers
       class Violation
-        attr_reader :issue
-
         def initialize(language_strategy, issue, hashes)
           @language_strategy = language_strategy
-          @issue = issue
           @hashes = hashes
+          @issue = issue
         end
 
-        def format
-          {
-            "type": "issue",
-            "check_name": check_name,
-            "description": description,
-            "categories": ["Duplication"],
-            "location": format_location,
-            "remediation_points": calculate_points,
-            "other_locations": format_other_locations,
-            "content": content_body,
-            "fingerprint": fingerprint
-          }
+        def occurrences
+          hashes.map.with_index do |sexp, i|
+            Issue.new(language_strategy: language_strategy, check_name: check_name, current_sexp: sexp, other_sexps: other_sexps(hashes.dup, i))
+          end
         end
 
         def report_name
           "#{current_sexp.file}-#{current_sexp.line}"
         end
 
-        private
-
-        attr_reader :language_strategy, :hashes
-
-        def current_sexp
-          @location ||= sorted_hashes.first
+        def mass
```

Comment:
I don't think this method is actually needed on this class anymore? It's just relevant on `Issue` now, right?